vpr/src/server/telegramoptions.h

```diff
@@ -34,19 +34,83 @@ class TelegramOptions {
     };
 
 public:
+    /**
+     * @brief Constructs a TelegramOptions object with the provided data and expected keys.
+     *
+     * This constructor initializes a TelegramOptions object with the given data string
+     * and a vector of expected keys. It parse data string and validate that such data has all required keys.
+     * If some keys are absent, it collects the errors.
+     *
+     * @param data The data string containing the options.
+     * @param expected_keys A vector of strings representing the expected keys in the options.
+     */
     TelegramOptions(const std::string& data, const std::vector<std::string>& expected_keys);
     ~TelegramOptions()=default;
 
+    /**
+     * @brief Checks if there are any errors present.
+     *
+     * This function returns true if there are errors present in the error container,
+     * otherwise it returns false.
+     *
+     * @return True if there are errors present, false otherwise.
+     */
     bool has_errors() const { return !m_errors.empty(); }
 
-    std::map<std::size_t, std::set<std::size_t>> get_map_of_sets(const std::string& name);
-
-    std::string get_string(const std::string& name);
-
-    int get_int(const std::string& name, int fail_value);
-
-    bool get_bool(const std::string& name, bool fail_value);
-
+    /**
+     * @brief Retrieves a map of sets associated with the specified key.
+     *
+     * This function retrieves a map of sets associated with the specified key.
+     *
+     * @param key The key of the map of sets to retrieve.
+     * @return The map of sets associated with the specified key.
```

Comment:
What is in the sets?  I just see they are size_t values. What are they storing?

Reply:
fixed https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2560/commits/c63477551d322e48f60ca75a921a9bd86d76edd8